src/main/scala/com/github/codelionx/dodo/actors/ODMaster.scala

```diff
@@ -98,32 +107,69 @@ class ODMaster(nWorkers: Int, resultCollector: ActorRef, systemCoordinator: Acto
           (map, ind) => map + (ind -> orderEquivalencies(ind))
         )
         resultCollector ! OrderEquivalencies(
-          equivalenceClasses.map{ case (key, cols) =>
+          equivalenceClasses.map { case (key, cols) =>
             table(key).name -> cols.map(table(_).name)
           }
         )
+        log.info("Generating first candidates and starting search")
         odsToCheck ++= generateFirstCandidates(reducedColumns)
+
+        // sending work to idleWorkers
+        if (odsToCheck.isEmpty) {
+          log.error("No OCD candidates generated!")
+          systemCoordinator ! Finished
+
+        } else {
+          val queueLength = odsToCheck.length
+          val workers = idleWorkers.length
+          val batchLength = math.min(
+            queueLength / workers,
```

Comment:
this throws an error when there are not idle workers, as the program tries to divide by zero. 
Maybe use `else if (idleWorkers.nonEmpty)` or something similar (line 122)

Reply:
Thank you. Done.